numpy/core/src/npysort/quicksort.c.src

```diff
@@ -69,10 +70,13 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
     @type@ *stack[PYA_QS_STACK];
     @type@ **sptr = stack;
     @type@ *pm, *pi, *pj, *pk;
-    npy_intp depth_limit = npy_get_msb(num) * 2;
+    int depth_limit = NPY_SORT_MIN(PYA_QS_STACK, npy_get_msb(num) * 2);
+    int depth[PYA_QS_STACK];
+    int * psdepth = depth;
+    int cdepth = 0;
 
     for (;;) {
-        if (depth_limit-- < 0) {
+        if (cdepth >= depth_limit) {
```

Comment:
hm yes an unlikely makes sense here

to my knowledge the length of a jump does not really matter (if both locations are in cache) and location in code has very little influence on where code ends up after compilation